shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -856,9 +857,29 @@ - (instancetype)initWithOwner:(FlutterTextInputPlugin*)textInputPlugin {
     }
   }
 
+  if (@available(iOS 16.0, *)) {
+    _editMenuInteraction = [[UIEditMenuInteraction alloc] initWithDelegate:self];
+    [self addInteraction:_editMenuInteraction];
+  }
+
   return self;
 }
 
+- (UIMenu*)editMenuInteraction:(UIEditMenuInteraction*)interaction
+          menuForConfiguration:(UIEditMenuConfiguration*)configuration
+              suggestedActions:(NSArray<UIMenuElement*>*)suggestedActions API_AVAILABLE(ios(16.0)) {
+  return [UIMenu menuWithChildren:suggestedActions];
```

Comment:
So in future milestones the framework could pass UIMenuElements like this?

Reply:
Yes, the framework can send the items via `showNativeEditMenu` channel, and we can return it here.

Reply:
Custom actions (e.g. send email) can be a bit more involved (e.g. the engine would need to call back to the framework).